src/OmniSharp.Abstractions/Endpoints/IGotoDefinition.cs

```diff
@@ -0,0 +1,11 @@
+using System.Threading.Tasks;
+using OmniSharp.Models;
+
+namespace OmniSharp.Endpoints
+{
+    public interface IGotoDefintion
+    {
+        Task<bool> IsApplicableTo(GotoDefinitionRequest request);
```

Comment:
Thinking about this more, the IsApplicableTo logic shouldn't be here at all. It should be at the top level (what types of files does this plugin handle?)